subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

```diff
@@ -515,6 +515,20 @@ To help you understand how incremental compilation works, the following provides
 * Having a source structure that does not match the package names, while legal for compilation, might end up causing trouble in the toolchain.
     Even more if annotation processing and <<build_cache.adoc#build_cache,caching>> are involved.
 
+Additionally, Gradle might temporarily change the output location while running incremental compilation.
+This might affect some annotation processors that inspect output locations or accept file paths as arguments (e.g. `-XepExcludedPaths` in Error Prone).
+You can disable this behaviour by setting link:{javadocPath}/org/gradle/api/tasks/compile/CompileOptions.html#getIncrementalAfterFailure--[`options.incrementalAfterFailure`] to `false` on the JavaCompile task.
+Alternatively, you can try to specify a temporary output path to the annotation processor.
```

Comment:
```suggestion
Additionally, Gradle might temporarily change the output location while running incremental compilation. 
This might affect some annotation processors that inspect output locations or accept file paths as arguments (e.g., `-XepExcludedPaths` in Error Prone).
There are two options:
 - You can disable `incremental after failure` by setting the link:{javadocPath}/org/gradle/api/tasks/compile/CompileOptions.html#getIncrementalAfterFailure--[`options.incrementalAfterFailure`] to `false` on the JavaCompile task.
-  You can keep `incremental after failure` working by passing a temporary output value to the annotation processor as a parameter so the annotation processor is aware of that path. In case of Error Prone, pass a value to `-XepExcludedPaths`. For example:
Before, you might have had: `-XepExcludedPaths=.*/build/generated/.*"`. With the new behavior, they might need to do
`-XepExcludedPaths=.*/(build/generated|compileJava/compileTransaction/annotation-output)/.*"`.

Note the need for this argument is planned to be resolved in [#23226](https://github.com/gradle/gradle/pull/23226).
```

Reply:
not in love with it, feel free to change

Reply:
Looks good to me. I would just exclude the PR, since it is not yet complete and we might find some surprises.

Reply:
I removed the note and applied a suggestion